lib/routes/users/routes.js

```diff
@@ -59,5 +75,129 @@ module.exports = [
                 }
             }
         }
+    },
+    {
+        method: 'POST',
+        path: '/register',
+        options: {
+            description: 'register new user.',
+            notes: 'POST users API',
+            tags: ['api', 'users'],
+            cors: true,
+            auth: false,
+            validate: {
+                payload: Joi.object({
+                    firstName: stringSchema,
+                    lastName: stringSchema,
+                    mobileNo: stringSchema,
+                    email: emailSchema,
+                    gender: Joi.string().valid('male', 'female', 'other')
+                })
+            }
+        },
+        handler: async (request, h) => {
+            try {
+                let user = await registerUser(request.payload);
+                return h.response(user).code(200);
+            } catch (error) {
+                return h.response({ message: error.message }).code(400);
+            }
+        }
+    },
+    {
+        method: 'PUT',
+        path: '/profile/update',
```

Comment:
Let's not have multiple route paths. This will be `/users/profile/update` right? That's a little too big.

Reply:
Done.